src/Blocks/BlockCrops.h

```diff
@@ -23,7 +23,30 @@ class cBlockCropsHandler :
 	{
 		cFastRandom rand;
 
-		if (a_Meta == 0x7)
+		// Beetroots have three stages before fully grown
+		if (a_Meta == 0x3)
+		{
+			// Is fully grown, drop the entire produce:
+			switch (m_BlockType)
+			{
+				case E_BLOCK_BEETROOTS:
+				{
+					char SeedCount = static_cast<char>(1 + (rand.NextInt(3) + rand.NextInt(3)) / 2);
+					a_Pickups.push_back(cItem(E_ITEM_BEETROOT_SEEDS, SeedCount, 0));
+					char BeetrootCount = static_cast<char>(1 + (rand.NextInt(3) + rand.NextInt(3)) / 2);
+					a_Pickups.push_back(cItem(E_ITEM_BEETROOT, BeetrootCount, 0));  // [1 .. 3] with high preference of 2
+					break;
+				}
+				default:
+				{
+					ASSERT(!"Unhandled block type");
+					break;
+				}
```

Comment:
This will not work. What if the block is a carrot, but has a meta of 3? This block of code will assert, although it shouldn't.

The code needs restructuring. Best if you add a `m_MaxMeta` member to the `cBlockCropsHandler` class, initialize it in the constructor based on the block type, and then use it in the condition below (`a_Meta >= m_MaxMeta`) instead of `a_Meta == 0x7`) and add the `E_BLOCK_BEETROOTS` case there. Possibly even assert when `a_Meta >m_MaxMeta` (although this could happen when loading worlds from other software, so not sure if such an assert is wise.)